src/devices/machine/idehd.cpp

```diff
@@ -379,22 +379,22 @@ void ata_mass_storage_device::security_error()
 
 attotime ata_mass_storage_device::seek_time()
 {
-	int sectors_per_cylinder =  m_num_heads * m_num_sectors;
+	uint32_t sectors_per_cylinder =  m_num_heads * m_num_sectors;
 
 	if (sectors_per_cylinder == 0 || m_num_cylinders == 0)
 		return attotime::zero;
 
-	int new_lba = lba_address();
-	int old_cylinder = m_cur_lba / sectors_per_cylinder;
-	int new_cylinder = new_lba / sectors_per_cylinder;
-	int diff = abs(old_cylinder - new_cylinder);
+	uint32_t new_lba = lba_address();
+	uint16_t old_cylinder = m_cur_lba / sectors_per_cylinder;
+	uint16_t new_cylinder = new_lba / sectors_per_cylinder;
+	uint16_t diff = abs(old_cylinder - new_cylinder);
 
 	m_cur_lba = new_lba;
 
 	if (diff == 0)
 		return TIME_BETWEEN_SECTORS;
 
-	attotime seek_time = (TIME_FULL_STROKE_SEEK * diff) / m_num_cylinders;
+	attotime seek_time = ((TIME_FULL_STROKE_SEEK - TIME_PER_ROTATION) * m_num_cylinders) / diff;
```

Comment:
This doesn't seem to be right at all - if anything this is the reciprocal of what the seek time should be.
* `TIME_FULL_STROKE_SEEK` is the time to seek across the radius of the recording surface.
* Suppose the disk has 129 physical tracks, seeking by 128 tracks should take `TIME_FULL_STROKE_SEEK`, and seeking by 64 tracks should take `TIME_FULL_STROKE_SEEK / 2`.

So for the 128 track seek, with your formula, `m_num_cylinders` is 129 and `diff` is 128 (ignoring the `TIME_PER_ROTATION` adjustment for simplicity):
* `TIME_FULL_STROKE * m_num_cylinders / diff` = `TIME_FULL_STROKE * 129 / 128`
* Slightly more than `TIME_FULL_STROKE` - this can't be right.

For the 64 track seek, with your formula, `m_num_cylinders` is 129 and `diff` is 64 (ignoring the `TIME_PER_ROTATION` adjustment for simplicity:
* `TIME_FULL_STROKE * m_num_cylinders / diff` = `TIME_FULL_STROKE * 129 / 64`
* Almost *double* `TIME_FULL_STROKE` - this definitely can't be right.